.github/workflows/build-node-packages.yml

```diff
@@ -0,0 +1,202 @@
+name: Generate Node.js
+on:
+# TODO: currently workflow dispatch endpoint does not work. I will investigate
+  workflow_dispatch:
+    inputs:
+      VERSION:
+        description: 'Node version to build and upload'
+        required: true
+        default: '14.2.0'
+      PUBLISH_RELEASES:
+        description: 'Whether to publish releases'
+        required: true
+        default: 'false'
+
+env:
+  VERSION: ${{ github.event.inputs.VERSION }}
+  ARCHITECTURE: x64
+
+jobs:
+  build_node:
+    name: Build Node ${{ github.event.inputs.VERSION }} ${{ matrix.platform }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [linux, darwin, win32]
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Create artifact directories
+      run: |
+        binariesDirectory=$RUNNER_WORKSPACE/binaries
+        echo ::set-env name=BINARIES_DIRECTORY::$binariesDirectory
+        mkdir $binariesDirectory
+
+        artifactDirectory=$RUNNER_WORKSPACE/artifact
+        echo ::set-env name=ARTIFACT_DIRECTORY::$artifactDirectory
+        mkdir $artifactDirectory
+
+    - name: Build Node ${{ env.VERSION }}
+      run: |
+        ./builders/build-node.ps1 -Version $env:VERSION `
+                                  -Platform ${{ matrix.platform }} `
+                                  -Architecture $env:ARCHITECTURE
+      shell: pwsh
+
+    - name: Publish artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: node-${{ env.VERSION }}-${{ matrix.platform }}
+        path: /home/runner/work/node-versions/artifact
+
+  test_node:
+    name: Test Node ${{ github.event.inputs.VERSION }} ${{ matrix.platform }}
+    needs: build_node
+    runs-on: ${{ matrix.os }}-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - os: ubuntu
+          platform: linux
+        - os: macos
+          platform: darwin
+        - os: windows
+          platform: win32
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Set AGENT_TOOLSDIRECTORY variable
+      if: matrix.platform == 'win32'
+      run: |
+        # GitHub Windows images don't have `AGENT_TOOLSDIRECTORY` variable
+        echo ::set-env name=AGENT_TOOLSDIRECTORY::$RUNNER_TOOL_CACHE
```

Comment:
How setup-node works on GitHub Windows machines in this case?

Reply:
discussed